com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStepTest.java

```diff
@@ -154,10 +173,16 @@ public void setUp() {
         context.setVariable(Constants.PARAM_NO_START, false);
         when(appStateCalculator.computeCurrentState(any())).thenReturn(currentAppState);
         when(appStateCalculator.computeDesiredState(any(), eq(false))).thenReturn(desiredAppState);
-        step.appStateCalculatorSupplier = () -> appStateCalculator;
-        CloudApplicationExtended app = new CloudApplicationExtended(null, "dummy");
+        step.appStateCalculatorSupplier = ApplicationStartupStateCalculator::new;
+        CloudApplicationExtended app = new CloudApplicationExtended(null, DUMMY);
         app.setRestartParameters(appRestartParameters);
         context.setVariable(Constants.VAR_APP_TO_DEPLOY, JsonUtil.toJson(app));
+        when(execution.getControllerClient()).thenReturn(client);
```

Comment:
This line should not be necessary, if you reuse the client mock from the parent class.

Reply:
Removed